include/igraph_memory.h

```diff
@@ -29,9 +29,13 @@
 
 __BEGIN_DECLS
 
-#define igraph_Calloc(n,t)    (t*) calloc( (size_t)(n), sizeof(t) )
-#define igraph_Realloc(p,n,t) (t*) realloc((void*)(p), (size_t)((n)*sizeof(t)))
-#define igraph_Free(p)        (free( (void *)(p) ), (p) = NULL)
+#define IGRAPH_CALLOC(n,t)    (t*) calloc( (n) > 0 ? (size_t)(n) : (size_t)1, sizeof(t) )
+#define IGRAPH_REALLOC(p,n,t) (t*) realloc((void*)(p), (n) > 0 ? (size_t)((n)*sizeof(t)) : (size_t)1)
```

Comment:
There is a slight discrepancy here; `IGRAPH_CALLOC()` and the earlier version of `igraph_Realloc()` ensured that the size of the allocated memory was a multiple of `sizeof(t)`. `IGRAPH_CALLOC()` ensures that in the new version as well, but `IGRAPH_REALLOC()` does not. I don't know whether that's a concern; probably not, but I'd like a second opinion here. (@szhorvat ?)

Reply:
I think  it shouldn't be a problem, after all this is memory that we (should!) never use.

I am not too experienced with alignment issues, but I don't see how this could cause trouble ...